python/py-graph-tool/Portfile

```diff
@@ -123,12 +96,11 @@ if {${name} ne ${subport}} {
         --with-expat-inc=${prefix}/include \
         --with-expat-lib="-L${prefix}/lib -lexpat"
 
-    # Clang uses the old libstc++ from gcc 4.2 before OS X 10.9. Boost doesn't
+    # Clang uses the old libstdc++ from gcc 4.2 before OS X 10.9. Boost doesn't
     # include some of the tr1 headers in libstdc++ and defines its own tr1
     # classes. This causes conflicts with sparsehash which insists on using
     # the old tr1 headers.
-    if {[string match *gcc* ${configure.compiler}] ||
-        ${os.major} >= 13 && ${os.platform} eq "darwin"} {
+    if {${configure.cxx_stdlib} ne "libstdc++"} {
         depends_lib-append port:sparsehash
     } else {
         configure.args-append --disable-sparsehash
```

Comment:
This attempts to address https://trac.macports.org/ticket/59890 but I am not certain I have done so correctly.

Reply:
I'm not sure either, mainly because I don't understand the comments there. If we're saying it works with libc++ but not with libstdc++, then you should check `if {${configure.cxx_stdlib} eq "libc++"}` (because `macports-libstdc++` is another valid value for `configure.cxx_stdlib`). The sparsehash port is intended to work with any C++ stdlib but maybe there is or was some special interaction between sparsehash and py-graph-tool that made that combination not work with libstdc++. The only systems that don't use libc++ by default are 10.4 and 10.5 and I can't get py-graph-tool or most anything else to build on 10.5 anymore because of https://trac.macports.org/ticket/57911. Should we maybe just remove the condition entirely, always enable sparsehash, and see if something breaks?